apiserver/application/application_unit_test.go

```diff
@@ -22,49 +23,46 @@ import (
 
 type ApplicationSuite struct {
 	testing.IsolationSuite
-	backend     mockBackend
-	application mockApplication
-	charm       mockCharm
-	endpoints   []state.Endpoint
-	relation    mockRelation
+	backend      mockBackend
+	applications map[string]*mockApplication
+	charms       map[string]*mockCharm
+	endpoints    []state.Endpoint
+	relation     mockRelation
 
 	blockChecker mockBlockChecker
 	authorizer   apiservertesting.FakeAuthorizer
-	api          *application.API
+	api          *application.APIv5
 }
 
 var _ = gc.Suite(&ApplicationSuite{})
 
+func (s *ApplicationSuite) TearDownTest(c *gc.C) {
+	for _, a := range s.applications {
```

Comment:
This implies that we are leaking applications between tests. s.application should start out empty or with newly cnstructed mocks.